Cubical/Algebra/Localisation.agda

```diff
@@ -0,0 +1,214 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+module Cubical.Algebra.Localisation where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Logic
+open import Cubical.Foundations.Transport
+open import Cubical.Functions.FunExtEquiv
+
+import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Bool
+open import Cubical.Data.Nat hiding (_+_ ; +-comm ; +-assoc)
+open import Cubical.Data.Vec
+open import Cubical.Data.Sigma.Base
+open import Cubical.Data.FinData
+open import Cubical.Relation.Nullary
+
+open import Cubical.Structures.Group hiding (⟨_⟩)
+open import Cubical.Structures.AbGroup hiding (⟨_⟩)
+open import Cubical.Structures.Monoid hiding (⟨_⟩)
+open import Cubical.Structures.Ring hiding (⟨_⟩)
+open import Cubical.Structures.CommRing
+
+open Iso
+
+private
+  variable
+    ℓ ℓ' : Level
+    A : Type ℓ
+
+record isSubMonoid (R' : CommRing {ℓ}) (S : ℙ ⟨ R' ⟩) : Type ℓ where
+ constructor
+   submonoid
+ field
+   containsOne : (R' .CommRing.1r) ∈ S
+   multClosed : ∀ {s t} → s ∈ S → t ∈ S → (R' .CommRing._·_ s t) ∈ S
+
+module _(R' : CommRing {ℓ}) (S : ℙ ⟨ R' ⟩) (SsubMonoid : isSubMonoid R' S) where
+ open isSubMonoid
+ open CommRing R' renaming (Carrier to R)
+ open Theory (CommRing→Ring R')
+
+
+ data S⁻¹R : Type ℓ where
+  _/_ : (r s : R) → {s ∈ S} → S⁻¹R
+  zd : (r₁ r₂ s s₁ s₂ : R) → {s ∈ S} → {a : s₁ ∈ S} → {b : s₂ ∈ S}
+     → s · ((r₁ · s₂) - (r₂ · s₁)) ≡ 0r
+     → ((r₁ / s₁) {a}) ≡ ((r₂ / s₂) {b})
```

Comment:
It's annoying to give `{a}` and `{b}` like this. Could one use instance arguments or something to have Agda infer them automatically? Maybe @Saizan or @guillaumebrunerie knows?

@mzeuner : here's the relevant documentation: https://agda.readthedocs.io/en/v2.6.0.1/language/instance-arguments.html

Reply:
It works quite well with instance arguments, the only time they were explicitly required was in the definition of addition:
[https://github.com/agda/cubical/pull/399/files#diff-bea1144102447f41fb1100cb7dd8f98fR157](url)

Thanks!